cmd/createRuleWaiver.go

```diff
@@ -85,15 +110,30 @@ func NewCreateRuleWaiverCommand() *cobra.Command {
 				fmt.Println(tableRow)
 			}
 		},
+		Args: func(cmd *cobra.Command, args []string) error {
+			// validate 'resource-id', 'resource-type', 'resource-provider' and 'resource-tag' when 'wildcard-mode=false'
+			if !opts.WildcardMode && (containsWildcards(opts.ResourceID) ||
+				containsWildcards(opts.ResourceType) ||
+				containsWildcards(opts.ResourceProvider) ||
+				containsWildcards(opts.ResourceTag)) {
+				return errors.New("'wildcard-mode=false' must only be used for exact matches. No wildcards are allowed")
+			}
+			if !isValidTag(opts.ResourceTag) {
+				return errors.New("'resource-tag' must be of the form 'key:value'")
+			}
+			return nil
```

Comment:
Just curious - would relying on server-side validation be an option here? Having the client simply pass through the values and asking the server to validate could make some sense.

Reply:
What do you suggest? Moving this sort of validation to the API?
Right now, the validation is the API is pretty minimal and this sort of validation is not implemented.

Reply:
It does seem like the kind of check that should happen in the API, yeah. What do you think @mattm-fugue?

Reply:
I'm in favor of removing the top check all together. API consumers may want to create exact match waivers with those characters, if we do eventually see them in resource IDs, so we should not prevent them from doing that.

The bottom check is better served in the API, but again, I wonder if it needs to exist. Customers could waive `key-prefix-*` and that would still work for tags like `key-prefix-a: foo` and `key-prefix-b: bar`